File: server/makc2022--nodejs--nestjs/src/layer2/nosql-mongo/queries/item/get/item-get-query-input.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from '../../../../../layer1/common/common-functions';
import { QueryInput } from '../../../../../layer1/query/query-input';

/** Входные данные запроса на получение элемента.
 */
export abstract class ItemGetQueryInput extends QueryInput {
  /** Идентификатор сущности. */
  entityId: string;

  /** @inheritdoc */
  override getInvalidProperties(): string[] {
    const result = super.getInvalidProperties();

    if (this.entityId === '') {
      result.push(nameof<ItemGetQueryInput>('entityId'));
    }

    return result;
  }

  /** Нормализовать. */
  normalize(): void {
    if (!this.entityId) {
      this.entityId = '';
    }
  }
}

File: server/makc2022--dotnet/Makc2022.Layer2.Sql/Queries/Item/Get/ItemGetQueryInput.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer1.Query;

namespace Makc2022.Layer2.Sql.Queries.Item.Get
{
    /// <summary>
    /// Входные данные запроса на получение элемента.
    /// </summary>
    public class ItemGetQueryInput : QueryInput
    {
        #region Properties

        /// <summary>
        /// Идентификатор сущности.
        /// </summary>
        public long EntityId { get; set; }

        #endregion Properties

        #region Public methods

        /// <summary>
        /// Нормализовать.
        /// </summary>
        public virtual void Normalize()
        {
            if (EntityId < 0)
            {
                EntityId = 0;
            }
        }

        /// <inheritdoc/>
        public override List<string> GetInvalidProperties()
        {
            var result = base.GetInvalidProperties();

            if (EntityId < 1)
            {
                result.Add(nameof(EntityId));
            }

            return result;
        }

        #endregion Public methods
    }
}
